src/phoenix/trace/dsl/query.py

```diff
@@ -39,60 +52,29 @@
     "trace_id": "context.trace_id",
 }
 
-# Because span_kind is an enum, it needs to be converted to string,
-# so it's serializable by pyarrow.
-_CONVERT_TO_STRING = ("span_kind",)
-
 
 def _unalias(key: str) -> str:
     return _ALIASES.get(key, key)
 
 
 @dataclass(frozen=True)
-class Projection:
-    key: str = ""
-    value: Callable[[Span], Any] = field(init=False, repr=False)
-    span_fields: ClassVar[Tuple[str, ...]] = tuple(f.name for f in fields(Span))
-
-    def __bool__(self) -> bool:
-        return bool(self.key)
+class _Base:
+    """The sole purpose of this class is for `super().__post_init__()` to work"""
 
     def __post_init__(self) -> None:
-        key = _unalias(self.key)
-        object.__setattr__(self, "key", key)
-        if key.startswith(CONTEXT_PREFIX):
-            key = key[len(CONTEXT_PREFIX) :]
-            value = partial(self._from_context, key=key)
-        elif key.startswith(ATTRIBUTE_PREFIX):
-            key = self.key[len(ATTRIBUTE_PREFIX) :]
-            value = partial(self._from_attributes, key=key)
-        elif key in self.span_fields:
-            value = partial(self._from_span, key=key)
-        else:
-            value = partial(self._from_attributes, key=key)
-        if self.key in _CONVERT_TO_STRING:
-            object.__setattr__(
-                self,
-                "value",
-                lambda span: None if (v := value(span)) is None else str(v),
-            )
-        else:
-            object.__setattr__(self, "value", value)
+        pass
```

Comment:
Is this needed if we just don't call `super().__post_init__()` in `_HasTmpSuffix`?

Reply:
I tried it, and in that case the `__post_init__` on Projection doesn't run

Reply:
huh, weird